Bottom: 4d16144df7594247494ac32e3b4405fa794c8a5e
Top:    1ccddb26440d65bd665ad03b5d9837912a54bce0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-04 12:26:26 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index a8dc5c0..194d279 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1163,7 +1163,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
